Doc/library/string.rst

```diff
@@ -755,8 +755,18 @@ attributes:
 
 * *idpattern* -- This is the regular expression describing the pattern for
   non-braced placeholders.  The default value is the regular expression
-  ``[_a-z][_a-z0-9]*``.  If this is given and *braceidpattern* is ``None``
-  this pattern will also apply to braced placeholders.
+  ``(?-i:[_a-zA-Z][_a-zA-Z0-9]*)``. Since default *flags* is
+  ``re.IGNORECASE``, ``[a-z]``Without local flag ``-i``, is used to avoid to match with non ASCII characters.
+  If this is given and *braceidpattern* is
+  ``None`` this pattern will also apply to braced placeholders.
+
+  .. note::
+
+     Default *flags* is ``re.IGNORECASE``.  So the pattern ``[a-z]`` can match
+     with some non ASCII characters.  That's why We use local ``-i`` flag here.
+
+     When overrinding this class, please consider overriding *flags* with ``0``
+     or ``re.IGNORECASE | re.ASCII``.
```

Comment:
How about: "When subclassing, please..."

Why?   Or in other words, can you add a short explanation of why they should consider this?